### PR TITLE
add sanity check on task ranks

### LINKED ISSUE #2655: Scheduler should fail on 0 ranks
Well, that should probably be caught earlier (in this case when submitting a raptor worker) - but either way, the scheduler should also fail.

### PR DESCRIPTION
**Pull request by andre-merzky**
This fixes #2655

### CODE REVIEW

File: src/radical/pilot/agent/scheduler/base.py [hunk 1]
```diff
@@ -790,6 +783,22 @@ def _schedule_waitpool(self):
...
 
 
+    # --------------------------------------------------------------------------
+    #
+    def _fail_task(self, task, e, detail):
+
+        task['control']          = 'tmgr_pending'
+        task['exception']        = repr(e)
+        task['exception_detail'] = detail
+        task['target_state']     = 'FAILED'
```
Comment (mtitov) [inline src/radical/pilot/agent/scheduler/base.py:793]: `task['target_state'] = rps.FAILED`

File: src/radical/pilot/agent/scheduler/base.py [hunk 2]
```diff
@@ -790,6 +783,22 @@ def _schedule_waitpool(self):
...
+        task['$all']             = True
+
+        self._log.exception('scheduling failed for %s', task['uid'])
+
+        self.advance(task, rps.FAILED, publish=True, push=False,
+                           fwd=True)
```
Comment (mtitov) [inline src/radical/pilot/agent/scheduler/base.py:799]: btw, for `AgentComponent`s we always have `fwd=True`, unless it is set explicitly `fwd=False`

**Review by mtitov — APPROVED** [review]
LGTM! 
(have just small comments)

**Comment by andre-merzky** [comment]
Thanks for both comments - applied the changes!

**Pull request merged**